src/library/Table/TableSortableHeaderCell.js

```diff
@@ -0,0 +1,198 @@
+    };
+  },
+  button: ({ theme: baseTheme }) => {
+    const theme = componentTheme(baseTheme);
+
+    return {
+      background: 'none',
+      border: 0,
+      color: 'inherit',
+      cursor: 'inherit',
+      fontSize: 'inherit',
+      fontWeight: 'inherit',
+      position: 'relative',
+      verticalAlign: theme.TableHeaderCell_verticalAlign,
+      whiteSpace: 'nowrap',
+      width: '100%',
+
+      '&:focus': focusStyles(theme),
+
+      '*:focus-within > &:focus': {
+        outline: 0
+      },
+
+      '&::-moz-focus-inner': {
+        border: 0
+      },
+
+      // [1]
+      '&::after': {
+        bottom: '100%',
+        content: '""',
+        height: '10em',
+        left: 0,
+        position: 'absolute',
+        right: 0
+      }
+    };
+  },
+  content: {
+    whiteSpace: 'normal'
+  },
+  iconHolder: ({ isActiveSort, direction, theme }) => {
+    const iconAdjustment = pxToEm(2);
+    const space = `${parseFloat(theme.space_inline_xxs) +
+      parseFloat(iconAdjustment)}em`;
+
+    return {
+      color: theme.icon_color,
+      display: 'inline-block',
+      height: '0.875em',
+      marginLeft: theme.direction === 'ltr' ? space : null,
+      marginRight: theme.direction === 'rtl' ? space : null,
+      opacity: isActiveSort ? null : 0,
+      position: 'relative',
+      top: direction === 'ascending' ? 2 : 1,
+      width: '0.875em',
+
+      '& > [role="img"]': {
+        margin: `-${iconAdjustment}`
+      },
+
+      '*:hover > button > &, button:focus > &': {
+        color: 'inherit',
+        opacity: 1
+      }
+    };
+  }
+};
+
+const Root = createStyledComponent(TableHeaderCell, styles.root, {
+  displayName: 'TableSortableHeaderCell',
+  withProps: { noPadding: true }
+});
+const Button = createStyledComponent(TableHeaderCell, styles.button, {
+  withProps: { element: 'button' }
+});
+const Content = createStyledComponent('span', styles.content);
+const IconHolder = createStyledComponent('span', styles.iconHolder);
+
+const iconProps = {
+  'aria-hidden': true,
+  size: 'auto'
+};
+const sortIcon = {
+  ascending: <IconArrowDropdownUp {...iconProps} />,
+  descending: <IconArrowDropdownDown {...iconProps} />
+};
+
+export default function TableSortableHeaderCell({
+  children,
+  label,
+  name,
+  messages,
+  sortable,
+  ...restProps
+}: Props) {
+  const { sort, sortFn } = sortable;
+
+  const sortColumn = sort && sort.key;
+  const descending = sort ? sort.descending : false;
+
+  const isActiveSort = sortColumn === name;
+  const activeDirection = descending ? 'descending' : 'ascending';
+  const nextDirection =
+    isActiveSort && activeDirection === 'ascending'
+      ? 'descending'
+      : 'ascending';
+
+  const toggle = isActiveSort || activeDirection !== nextDirection;
+  const nextDescending = toggle ? !descending : descending;
+
+  const a11yLabel = label || children;
+
+  const rootProps = {
+    ...restProps,
+    // TODO: Mac Chrome VO (others?) announces this twice?
```

Comment:
Not sure if you got through a11y testing yet, but reminder to update/remove this comment